ledger/internal/double_test.go

```diff
@@ -145,7 +145,22 @@ func checkBlock(t *testing.T, checkLedger *ledger.Ledger, vb *ledgercore.Validat
 	cb := endBlock(t, checkLedger, check)
 	check.SetGenerate(false)
 	require.Equal(t, vb.Block(), cb.Block())
-	require.Equal(t, vb.Delta(), cb.Delta())
+
+	// vb.Delta() need not actually be Equal, in the sense of require.Equal
```

Comment:
@jannotti Gauging if it's worth further investigation - Can you take a look at what I'm seeing and let me know if you feel more effort is warranted?

https://github.com/algorand/go-algorand/pull/3896/commits/e88ae3dfb3853fb6ec5a2d2a64f835b7c37e129f caught my attention and I began wondering what alternatives exist.  If you feel the PR's approach is robust enough, then I'll happily close the thread.

* My concern is drift between `StateDelta` and the test assertions will widen over time.  For example, imagine defining `StateDelta` fields and forgetting to update the test.
* Brief research suggests a couple of options:  https://stackoverflow.com/questions/18208394/how-to-test-the-equivalence-of-maps-in-golang.  
* From my surface-level review, https://github.com/google/go-cmp feels like the most principled way to address the concern.    

Aside:  How did you discover the map ordering issue?  
  * I originally thought there's some non-determinism in the tests resulting in periodic failures.  However, I ran `gotestsum --  ./ledger/internal` several times without producing a failure.

Reply:
There _was_ non determinism in the test, when it compared Delta(), but it was hard to tease out, because require.Equal fails miserably when trying to tell you about a difference in an object where the difference is in an unexported field. So I a) Changed the test code to run 100 times, ensuring a failure, and b) temporarily changed the innards of AccountDelta to be exported, so that I could actually see what was failing.

Reply:
As for comparing maps, that's not _quite_ the problem here.  require.Equal does a fine job for that, as far as I'm away.  The trouble is in the details of AccountDelta.  It uses to intertwined objects, a slice and a map.  The kind of equality we would want to check is that given to AccountDeltas, X and Y, for all keys k in X, X.slice[X.map[k]] is equal to Y.slice[X.map[k]] (and the size of the maps are the same).  So the definition of equality here is beyond and simple definition provided by DeepEquals.

Reply:
I _might_ be able to iterate the modified accounts, and lookup each possible thing (account itself, asset, app) and check them for equality, using exported methods.

Reply:
> Arguably, the solution is to define func (AccountData) Equivalent(AccountData) but such a method would have no use in real code, and can't be written to be exported only for testing.

@jannotti Does the lack of test-only visibility preclude the solution?

* Based on your reply and my understanding, I feel like I'd also try to define the needed functionality closer to the source object (like the quoted text).
* Although the visibility is imperfect, I don't immediately see the harm in having the method available.  Particularly because (I _think_) it's side-effect free. 
* Aside:  I don't know if helps the situation or if considered excessive, but since we're already touching on https://github.com/google/go-cmp:
  * Perhaps the equivalence method in-question is more explicitly delineated as test-only by defining a go-cmp specific comparison method _rather than_ a standard interface method?
  * Untested, only done reading:  go-cmp provides facilities to test unexported fields - https://pkg.go.dev/github.com/google/go-cmp/cmp#Exporter.  It's _not_ my preference to suggest a reflection-based approach.  I present it in the spirit of exploring options.

Reply:
Resolved offline - decided not to check Delta() at all, as the creation of the Delta() from a given block is outside the scope of this test.  If we get identical blocks, that should be sufficient.